jni/src/faiss_wrapper.cpp

```diff
@@ -367,14 +390,14 @@ void knn_jni::faiss_wrapper::CreateByteIndexFromTemplate(knn_jni::JNIUtilInterfa
     jbyte * indexBytesJ = jniUtil->GetByteArrayElements(env, templateIndexJ, nullptr);
 
     faiss::VectorIOReader vectorIoReader;
+    vectorIoReader.data.reserve(indexBytesCount);
```

Comment:
why we need this?

Reply:
To avoid frequent malloc calls. It's an old trick.
Default implementation of vector uses doubling strategy when it reaches its max capacity, and given that we already know the size we are going to use, then we can just allocate the required memory without having to have it grew gradually up to the point.

Reply:
I understand the part of `vector uses doubling strategy when it reaches its max capacity`, may I was not clear on the question, so it was mainly trying to ask here is where is this `vectorIoReader.data` being used basically what it is for? 

So what I can see from here is, we are creating an indexWithTemplate(so indices like IndexIVF and PQ based indices is being created here). We are first putting all the data in the temporary buffer of `vectorIoReader` and then creating the index. I also see that we are doing the same thing where we are ingesting the same vectors via `faiss::IndexIDMap idMap`. This does seems weird to me. 

@jmazanec15  any thoughts here?

Reply:
Thanks @jmazanec15 and @0ctopus13prime for the details. I got my answer. Didn't want to hold off on this PR. Approving the PR.